src/devices/bus/msx/beecard/beecard.cpp

```diff
@@ -0,0 +1,142 @@
+// license:BSD-3-Clause
+// copyright-holders:Wilbert Pol
+
+#include "emu.h"
+#include "beecard.h"
+
+
+DECLARE_DEVICE_TYPE(BEE_CARD_NOMAPPER, bee_card_interface)
+
+
+void bee_card(device_slot_interface &device)
+{
+	device.option_add_internal("nomapper", BEE_CARD_NOMAPPER);
+}
+
+
+bee_card_interface::bee_card_interface(const machine_config &mconfig, device_t &device)
+	: device_interface(device, "beecard")
+	, m_page{nullptr, nullptr, nullptr, nullptr}
+{
+}
+
+void bee_card_interface::rom_alloc(u32 size)
+{
+	m_rom.resize(size);
+	std::fill_n(m_rom.begin(), size, 0xff);
+}
```

Comment:
This has no business being here.  The slot device should ensure the same regions are created whether media is loaded via the software list or from a loose file.  We should be *removing* these from media devices, not adding more devices with them.